CMakeLists.txt

```diff
@@ -24,34 +24,32 @@ LIST(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
 # Try using the PCH - it will only work if you have the right version of cmake
 #
 option(USING_PCH "Build using pre-compiled header support" ON)
+option(WITH_OPENMP "Use OpenMP for parallelization" ON)
+
+
+#
+#  OpenMP
+#
+if(WITH_OPENMP)
+  find_package(OpenMP REQUIRED)
+endif()
 
 
 #
 #  Python library
 #
-MESSAGE("Looking for Python libraries...")
-MESSAGE("Advice: If this search fails, try setting cache variables: -DPYTHON_LIBRARY=\${path_to_the_python_library} -DPYTHON_INCLUDE_DIR=\${path_to_where_Python.h_is_found}")
-MESSAGE("Hint to self: -DPYTHON_INCLUDE_DIR=/home/Alexey_2/Soft/Python-2.6.7/bin/include/python2.6 -DPYTHON_LIBRARY=/home/Alexey_2/Soft/Python-2.6.7/bin/bin")
-
-# For Linux and Cygwin
-# Adapt it according to your system
-#FIND_PACKAGE(PythonLibs 3.6 REQUIRED)
-FIND_PACKAGE(PythonLibs 3.9 REQUIRED)
-#FIND_PACKAGE(PythonLibs 3.7 REQUIRED)
-
-IF(PYTHONLIBS_FOUND)
-  MESSAGE("Success!")
-  INCLUDE_DIRECTORIES("${PYTHON_INCLUDE_DIRS}")
-ELSEIF(NOT PYTHONLIBS_FOUND)
-  MESSAGE("Unable to find correct Python version. If your Python include/libraries are installed in a non-standard location")
-  MESSAGE("Try using -DPYTHON_LIBRARY=\${path_to_the_python_library} -PYTHON_INCLUDE_DIR=\${path_to_where_Python.h_is_found}")
+MESSAGE(STATUS "Looking for Python libraries...")
+FIND_PACKAGE(Python3 3.6 REQUIRED COMPONENTS Interpreter Development)

```

Comment:
the version of Python can probably vary as long as that dev version of Python is installed; but i'd not be moving backwards to Python 3.6 in favor of 3.9 - why is that needed?

Reply:
Find_Package looks for the version, which is not lower than defined. I have installed Python3.9, 3.10, 3.11 and 3.12. The latest one was chosen.